src/IO/ITKImageInputFileFormat.cxx

```diff
@@ -185,7 +291,91 @@ read_file_itk(std::string filename)
       error(ex.what());
       return 0;
     }
+}
+
+//To read any file format via ITK
+template<>
+STIRImageMulti*
+read_file_itk(std::string filename)
+{
+#ifndef NDEBUG
+  info(boost::format("Opening a multi image: %1%") % filename);
+#endif
+
+  typedef itk::GDCMImageIO       ImageIOType;
+  ImageIOType::Pointer dicomIO = ImageIOType::New();
+
+  try
+    {
+      // We can't deal with DICOMs
+      if (dicomIO->CanReadFile(filename.c_str())) {
+          error("read_file_itk: multicomponent DICOM not implemented %s:%d.",
+                __FILE__, __LINE__);
+          return NULL;
+      }
+
+      // Not a DICOM file, so we just read a single image
+      typedef itk::ImageFileReader<ITKImageMulti> ReaderType;
+      ReaderType::Pointer reader = ReaderType::New();
+      reader->SetFileName(filename);
+      reader->Update();
+
+      if (reader->GetImageIO()->GetPixelType() != itk::ImageIOBase::VECTOR) {
```

Comment:
this doesn't do anything useful as far as I can see. It isn't a test on the `reader` really, it's a test on the `ITKImageMulti` type. But you have that under control. Remove this test.

Reply:
This is actually very important. If you don't set the `intent_code` to `VECTOR` when you write the nifti image, then ITK will read the image differently.

I made this mistake before, and although the ITK reader didn't complain, the images weren't read in the way that I expected.

Reply:
yes, but I believe (but didn't check) that this test will always be true as you're using `itk::ImageFileReader<itk::VectorImage<float, 3> >`. Presumably, the test on `intent_code` for nifti sits in the ITK instantiation for vector images (or it should).